setup.py

```diff
@@ -2,10 +2,23 @@
 
 import setuptools
 
+test_requires = [
+    "pytest>=3.7.1",
+    "pytest-cov>=2.5.1",
+    "tox>=3.2.1",
+    "flake8>=3.7.9",
+    "black>=19.3b0",
+    "pre_commit>=2.10.1",
+]
+
+extras_require = {
+    "test": test_requires,
```

Comment:
WDYT about using a `dev` extra instead of `test`? This would be more in line with other package managers like [poetry](https://python-poetry.org/docs/cli/#options_3) or [npm](https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file)

Reply:
yes, I would like to separate `test` and `dev`. Something like this:
```python
dev_requires = test_requires + ["pre_commit>=2.10.1", "black"]

extras_require = {
    "dev": dev_requires,
    "test": test_requires,
    "docs": coverage_requires,
}
```

How do people feel about this? @zapatacomputing/quantum-software

Reply:
Not sure why `"docs": coverage_requires` and not `"docs": docs_requires`, but in general this looks fine to me.

Reply:
@mstechly oops! I had separate docs, build, test, coverage, dev originally. It just just an example from an old project :P

Reply:
I agree it's a good idea to separate dependencies needed when we develop the library from the ones that a user needs to install when they only want to use the library in their project. 

Going farther with the granularity than just two options seems counter-productive for me, because this would require me to specify all the extras when I develop the library. Is there a reason we don't wanna do the same path as poetry and npm?

Reply:
tl;dr my suggestion is to use just a single, `dev` extra that would contain all development tools, like `pytest` or `black`